main/acle.md

```diff
@@ -9382,6 +9416,2341 @@ are named after. All of the functions have external linkage.
+
+Multi-vector clamp to minimum/maximum vector
+
+``` c
+  // Variants are also available for _f32, _f64, _s8, _s16, _s32, _s64, _u8, _u16,
+  // _u32 and _u64
+  __attribute__((arm_streaming_compatible))
+  svfloat16_t svclamp[_f16](svfloat16_t zd, svfloat16_t zn, svfloat16_t zm);
+  
+  
+  // Variants are also available for _f32, _f64, _s8, _s16, _s32, _s64, _u8, _u16,
+  // _u32 and _u64
+  __attribute__((arm_streaming))
+  svfloat16x2_t svclamp[_single][_f16][_x2](svfloat16x2_t zd, svfloat16_t zn,
+                                            svfloat16_t zm);
+  
+  
+  // Variants are also available for _f32, _f64, _s8, _s16, _s32, _s64, _u8, _u16,
+  // _u32 and _u64
+  __attribute__((arm_streaming))
+  svfloat16x4_t svclamp[_single][_f16][_x4](svfloat16x4_t zd, svfloat16_t zn,
+                                            svfloat16_t zm);
+  
+  ```
+
+#### REVD
+
+Reverse 64-bit doublewords in elements
+
+``` c
+  // Variants are also available for _u16, _u32 and _u64
+  __attribute__((arm_streaming_compatible))
+  svuint8_t svrevd[_u8]_m(svuint8_t zd, svbool_t pg, svuint8_t zn);
+  
+  
+  // Variants are also available for _u16, _u32 and _u64
+  __attribute__((arm_streaming_compatible))
+  svuint8_t svrevd[_u8]_x(svbool_t pg, svuint8_t zn);
+  
+  
+  // Variants are also available for _u16, _u32 and _u64
+  __attribute__((arm_streaming_compatible))
+  svuint8_t svrevd[_u8]_z(svbool_t pg, svuint8_t zn);
+  
+  ```
+
+#### SEL
+
+Multi-vector conditionally select elements from two vectors
+
+``` c
+  // Variants are also available for _u16, _u32 and _u64
+  __attribute__((arm_streaming))
+  svuint8x2_t svsel[_u8][_x2](svcount_t png, svuint8x2_t zn, svuint8x2_t zm);
+  
+  
+  // Variants are also available for _u16, _u32 and _u64
+  __attribute__((arm_streaming))
+  svuint8x4_t svsel[_u8][_x4](svcount_t png, svuint8x4_t zn, svuint8x4_t zm);
+  
+  ```
+
+#### URSHL, SRSHL (single)
+
+Multi-vector rounding shift left
+
+``` c
+  // Variants are also available for _s16, _s32, _s64, _u8, _u16, _u32 and _u64
+  __attribute__((arm_streaming))
+  svint8x2_t svrshl[_single][_s8][_x2](svint8x2_t zdn, svint8_t zm);
+  
+  
+  // Variants are also available for _s16, _s32, _s64, _u8, _u16, _u32 and _u64
+  __attribute__((arm_streaming))
+  svint8x4_t svrshl[_single][_s8][_x4](svint8x4_t zdn, svint8_t zm);
+  
+  ```
+
+#### URSHL, SRSHL (multi)
+
+Multi-vector rounding shift left
+
+``` c
+  // Variants are also available for _s16, _s32, _s64, _u8, _u16, _u32 and _u64
+  __attribute__((arm_streaming))
+  svint8x2_t svrshl[_s8][_x2](svint8x2_t zdn, svint8x2_t zm);
+  
+  
+  // Variants are also available for _s16, _s32, _s64, _u8, _u16, _u32 and _u64
+  __attribute__((arm_streaming))
+  svint8x4_t svrshl[_s8][_x4](svint8x4_t zdn, svint8x4_t zm);
+  
+  ```
+
+#### SQRSHR, UQRSHR
+
+Multi-vector saturating rounding shift right narrow
+
+``` c
+  
+  __attribute__((arm_streaming))
+  svint16_t svqrshr[_s16][_x2](svint32x2_t zn, uint64_t imm_1_8);
```

Comment:
The immediate operand for the "multi-vector saturating rounding shift right" intrinsics are all named _imm_1_8_, though I believe the immediates are all in the range of 1 to number of bits per source element for these instructions. Should the name be changed to reflect this?

Reply:
Fair point. I've generalised the name to just `imm` to avoid the issue entirely.